vimeo-networking/src/main/java/com/vimeo/networking/utils/ISO8601Wrapper.java

```diff
@@ -0,0 +1,55 @@
+package com.vimeo.networking.utils;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import com.google.gson.internal.bind.util.ISO8601Utils;
+import com.vimeo.networking.logging.ClientLogger;
+
+import java.lang.reflect.Type;
+import java.text.ParseException;
+import java.text.ParsePosition;
+import java.util.Date;
+
+/**
+ * A wrapper class that serializes and deserializes dates
+ * using the {@link ISO8601Utils} class that catches errors
+ * and reports them to the client application, allowing
+ * them to decide if they should crash or not. If we just
+ * rely on the default adapter used by JSON, we are unable
+ * to absorb date parsing errors or log them correctly.
```

Comment:
Also it wasn't just logging them incorrectly.....it was serializing it incorrectly.